mlflow/tracing/processor/mlflow.py

```diff
@@ -69,9 +69,20 @@ def on_start(self, span: OTelSpan, parent_context: Optional[Context] = None):
         span._start_time = time.time_ns()
 
     def _start_trace(self, span: OTelSpan) -> TraceInfo:
+        metadata = {}
+
+        # If the span is started within an active MLflow run, we should record it as a trace tag
+        run_experiment_id = None
+        if run := mlflow.active_run():
+            metadata[TraceMetadataKey.SOURCE_RUN] = run.info.run_id
+            run_experiment_id = run.info.experiment_id
+
         experiment_id = (
-            get_otel_attribute(span, SpanAttributeKey.EXPERIMENT_ID) or _get_experiment_id()
+            run_experiment_id
+            or get_otel_attribute(span, SpanAttributeKey.EXPERIMENT_ID)
```

Comment:
Can we just use one variable
```suggestion
            experiment_id = None
            if run := mlflow.active_run():
                experiment_id = run.info.experiment_id
```

Reply:
oh yeah we don't even have to pre-assign to none in that case, i can just do this check after the original experiment_id assignment